torch/csrc/jit/passes/autocast.cpp

```diff
@@ -105,7 +133,7 @@ void castInputsToWidestType(Node* node) {
   for (auto input : node->inputs()) {
     if (auto tensor_type = input->type()->cast<TensorType>()) {
       const auto dtype = tensor_type->scalarType();
-      if (!dtype.has_value() || *dtype != at::ScalarType::Half) {
+      if (!dtype.has_value() || *dtype == at::ScalarType::Float) {
```

Comment:
This looks good to me but it's unrelated to the set_opt_dtype changes right?

Reply:
Correct. This is a small tweak I promised @jjsjann123 in a previous review.